pytest_odoo.py

```diff
@@ -98,6 +98,13 @@ def enable_odoo_test_flag():
     yield
     odoo.tools.config['test_enable'] = False
 
+@pytest.fixture
+def env():
+    dbname = odoo.tests.common.get_db_name()
+    registry = odoo.modules.registry.RegistryManager.get(dbname)
+    cr = registry.cursor()
+    uid = odoo.SUPERUSER_ID
+    return odoo.api.Environment(cr, uid, {})
```

Comment:
You should surround the environment with a `odoo.api.Environment.manage():` context manager:

```python
with odoo.api.Environment.manage():
    yield odoo.api.Environment(cr, uid, {})
```

Reply:
@guewen I think this fixture is pretty usefull for bdd with pytest-bdd, want me to re-open this with this?:

```python
@pytest.fixture(scope='session', autouse=True)
def env():
    db_name = odoo.tests.common.get_db_name()
    registry = odoo.registry(db_name)
    cr = registry.cursor()
    with odoo.api.Environment.manage():
        yield odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
    registry.clear_caches()
    registry.reset_changes()
    env.reset()
    cr.rollback()
    cr.close()
```

Reply:
If it has `autouse=True`, it means it will always be used right? Can you explain me what does it change compared to the current behavior and what would this fixture bring (which use case?)?

Reply:
Yes, but it will run only once.

I’m using it when i need the odoo enviroment and i’m not inheriting from odoo test clases, for example when doing integration tests cross-modules (in my case i’m using pytest-bdd, and i need it for a lot ir scenarios).

Else, you will always need to inherit from odoo test classes or set up this in every scenario.

What you think?